pkg/apis/coordination/types.go

```diff
@@ -67,3 +93,69 @@ type LeaseList struct {
 	// items is a list of schema objects.
 	Items []Lease
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// LeaseCandidate defines a candidate for a lease object.
+// Candidates are created such that coordinated leader election will pick the best leader from the list of candidates.
+type LeaseCandidate struct {
+	metav1.TypeMeta
+	// +optional
+	metav1.ObjectMeta
+	Spec LeaseCandidateSpec
+}
+
+// LeaseCandidateSpec is a specification of a Lease.
+type LeaseCandidateSpec struct {
+	// LeaseName is the name of the lease for which this candidate is contending.
+	// This field is immutable.
+	LeaseName string
+	// PingTime is the last time that the server has requested the LeaseCandidate
+	// to renew. It is only done during leader election to check if any
+	// LeaseCandidates have become ineligible. When PingTime is updated, the
+	// LeaseCandidate will respond by updating RenewTime.
+	// +optional
+	PingTime *metav1.MicroTime
+	// RenewTime is the time that the LeaseCandidate was last updated.
+	// Candidates are expected to update their renewTime periodically,
```

Comment:
am not quite sure what I would have to do as user of this API. Everything else below reads like I don't have to periodically update and just wait for PingTime to change.

Reply:
If the user does not update renewTime periodically, its difficult for us to determine the difference between a candidate that has gone away vs a candidate that is still active and watching. We do not really need re-election if candidates are not changed and the curent leader is happily renewing. I feel like gc conditions would be difficult to determine.

Reply:
Why not regularly set ping time? It's not good to leave this as undefined as "renew, but not too often". That's not a clear specification.

Reply:
Regularly setting ping time now, removed the line around asking users to renew pingTime.